### ISSUE TITLE
[Suggestion] Please consider using GTK for the client

### ISSUE THREAD

**Issue by TheEvilSkeleton** [ValveSoftware/steam-for-linux#7454]
#### Your system information

* Steam client version (build number or date): 1603992987 (Flatpak)
* Distribution: Fedora
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:

I know it's an unusual issue, but please hear me out.

I love the work you people have done to improve gaming in Linux, but to be honest, the Steam client integrates into Linux terribly.

Since Steam uses Electron for the client, there is no uniformity with other GTK (and Qt) applications. It would be a great addition to select the theme we want and have Steam automatically use the GTK theme.

Second, Wayland is being heavily pushed by KDE, GNOME and Red Hat. Electron doesn't support Wayland at all, so users using Wayland will still have to fallback to X11.

Third, Flatpak. I personally use Flatpak when Flatpak is available, and a lot of users use the Steam flatpak as well. Using GTK will not only make Steam integrate more into Flatpak, but it will also make it really easy to utilize Flatpak features like [Portals](https://github.com/flatpak/flatpak/wiki/Portals).

Fourth, open source. There are a lot of users using Steam, so there is no doubt that Steam will receive a lot of contribution by open sourcing the client.

I hope it is going to be considered. I would love to see Steam integrate more into Linux.

**Comment by 12buntu** [comment]
This would be a super useful feature. I feel like if a Valve started a community effort, the community would help

**Comment by dennis1248** [comment]
The Linux community has been joking about a GTK Steam client for ages. And I would find it absolutely fantastic if FOSS Steam Client would be made upon GTK.

The chance of this happening, I think is very small. But with Valve you never know..

My vision of a GTK Steam client would be a simplistic client following the GNOME human interface guidelines which can install games, launch games, provide the usually Linux/Proton configuration options, basic Steam news and chat functionality. Store functionality would be optional. Preferably no build-in web browser.

The benefits on this:
- It would perform significantly better then the Chromium-based Steam client.
- Less likely to have significant bugs due to the much smaller code base.
- It's being open source would allow the community to build functionality upon it. For example imagine Box86 integration allowing 32 bit Steam games to be easily played on ARM-based devices running Linux.
- Valve are the good guys and the hero of the Linux community. 

The downsides:
- It will be a lot of work to build and maintain an app written in a low-level programming language like C, Rust or Vala.

**Comment by TheEvilSkeleton** [comment]
> My vision of a GTK Steam client would be a simplistic client following the GNOME human interface guidelines which can install games, launch games, provide the usual Linux/Proton configuration options, basic Steam news and chat functionality. Store functionality would be optional. Preferably no build-in web browser.

Agreed. There's a program called [GameHub](https://github.com/tkashkin/GameHub), which is a client for platforms like Steam. It's in GTK, and it looks amazing because it integrates in the Linux desktop:

![image](https://user-images.githubusercontent.com/50847364/99573305-ed258800-29a3-11eb-86d0-3c0c69722238.png)

![image](https://user-images.githubusercontent.com/50847364/99573656-5c9b7780-29a4-11eb-908c-4bf42c079325.png)

**Comment by 12buntu** [comment]
> > My vision of a GTK Steam client would be a simplistic client following the GNOME human interface guidelines which can install games, launch games, provide the usual Linux/Proton configuration options, basic Steam news and chat functionality. Store functionality would be optional. Preferably no build-in web browser.
> 
> Agreed. There's a program called [GameHub](https://github.com/tkashkin/GameHub), which is a client for platforms like Steam. It's in GTK, and it looks amazing because it integrates in the Linux desktop:
> 
> ![image](https://user-images.githubusercontent.com/50847364/99573305-ed258800-29a3-11eb-86d0-3c0c69722238.png)
> 
> ![image](https://user-images.githubusercontent.com/50847364/99573656-5c9b7780-29a4-11eb-908c-4bf42c079325.png)

How is this too much different from Lutris? That's my solution for now, but I don't get store or other functionalities

**Comment by TheEvilSkeleton** [comment]
GameHub's goal is to have quick access to all your games in one place, whereas Lutris' goal is to have access to Windows programs in Linux with the help of user-written scripts. Lutris *happens* to have a feature to store all your games in one application.

Most people would prefer Lutris, because of the scripts written by users, and because it is more mature.

**Comment by bonkmaykrQ** [comment]
I second this. While I like steam's built in look, making it able to absorb the color scheme or theme of your KDE desktop for example, or able to move the menu buttons from the top left to anywhere depending on the system's title bar text position, would help it blend in more with desktops that have a specific palette.

**Comment by 12buntu** [comment]
> or able to move the menu buttons from the top left to anywhere depending on the system's title bar text position,
@bonkmaykrQ that's my biggest issue, all of my other apps use menu buttons on the left, and it really surprised me that Steam doesn't respect that, given that Valve is really the hero of the Linux gaming community

**Comment by bonkmaykrQ** [comment]
Yeah! I always wanted to be able to move the close/minimize buttons to the right and move my action bar to the center.

**Comment by EntityinArray** [comment]
Wow! Even 8GB of RAM is considered small nowadays, due to ugly web apps. Would love to have Steam GTK web-client

**Comment by bonkmaykrQ** [comment]
Haha, I had 8 gb of ram for a long time. Never had a need to upgrade but I did anyway, at 16gb now and just might move to 32 to make room for a KVM. That being said, Electron isn't exactly the most resource efficient! GTK should have a smaller footprint.

**Comment by saltedcoffii** [comment]
+1 for a steam GTK client

**Comment by gbschenkel** [comment]
I would prefer a Qt/C++ or QtForPython approach instead.

**Comment by appetrosyan** [comment]
As a side note, as Chromium is essentially part of QtWebEngine, Steam could smoothly migrate using incremental steps that lead from a Chromium PWA, to a Qt native app, whilst retaining the code re-use that is apparent with respect to the Steam Store. 

Basically, the problem here is that the GTK client will only look good on Linux. Qt looks good everywhere.

**Comment by TheEvilSkeleton** [comment]
I've had far more problems with Qt than with GTK, especially with Flatpak and Wayland; thankfully KDE is working really hard on it. Anyway, if Qt is far easier to develop/maintain than GTK, then I guess I would be much happier than I am right now, especially how it look much better than whatever Steam is doing right now.

As for the "GTK client will only look good on Linux", I disagree. GIMP 2.99 has been looking great on Windows 10, and it's in GTK 3. https://www.youtube.com/watch?v=7aPDlxG9DUg

**Comment by 12buntu** [comment]
A GTK client would be nice, but I think that overall Qt might be a better choice. I don't have ~~much~~ any experience developing applications with Qt or GTK, so I can't speak to the difficulty of developing/maintaining the application, but I will say GTK tends to look better on Linux, at least on the DEs I use frequently.

**Comment by SeaDve** [comment]
I think it would be great to have a public API for steam, so third party developers can create frontends for gtk, qt etc.

**Comment by Faalagorn** [comment]
> I've had more problems with Qt than with GTK, especially with Flatpak and Wayland; thankfully KDE is working really hard on fixing those issues. Anyway, if Qt is easier to develop/maintain than GTK, then I wouldn't mind the switch to Qt instead, since the current client uses a lot more RAM than it should, especially when it's downloading and installing games/updates.
> 
> As for the "GTK client will only look good on Linux", I disagree. GIMP 2.99 for example has been looking great on Windows 10, and it's in GTK 3, https://www.youtube.com/watch?v=7aPDlxG9DUg. Maybe I'm cherry picking or being biased right now, but that's what I think. Though as I said, I think that even Qt would be marginally better than the current client.

Actually, GIMP (outside of development 3 version) is one of the last apps still stuck with GTK 2, which is deprecated: https://archlinux.org/todo/gtk-2-eol/

I agree with other options, I initially thought of Steam using GTK since it already has some dependences of it, but while both toolkits can look fine no matter the platform (though GTK has a feel of being more Linux-like due to its history indeed), Qt might be a better approach, as it's more Windows-like out of the box and Steam Deck will be using KDE which pusesh Qt heavily.

The look and feel (which got improved with the latest client updates, and more things can be done from web anyway), along the Wayland support #4924 (which is great on Qt) is the last major gripe I have. :) Open sourcing parts of it would surely help speed it up though, especially as other client does that (Itch, gamejolt and Desura prior to that).

**Comment by Cthatharsis** [comment]
Guys I found cool GTK  Adwaita theme for Steam. It looks neat and blends very well with other Adwaita apps. Link: [https://github.com/tkashkin/Adwaita-for-Steam](https://github.com/tkashkin/Adwaita-for-Steam)

![screenshot](https://user-images.githubusercontent.com/106710108/206826605-246a9fdd-82bc-455f-b1c0-d0e6c5b54704.png)

I know it's not GTK client but if you only care about how it loooks this might be good for you.

**Comment by werdahias** [comment]
there is https://crates.io/crates/steam-api for a rust-based approach. (I might be biased here but rust has nice gtk bindings)

**Comment by casanovalx** [comment]
> The Linux community has been joking about a GTK Steam client for ages. And I would find it absolutely fantastic if a FOSS Steam Client would be made upon GTK.
> 
> The chance of this happening, I think is very small. But with Valve you never know..
> 
> My vision of a GTK Steam client would be a simplistic client following the GNOME human interface guidelines which can install games, launch games, provide the usual Linux/Proton configuration options, basic Steam news and chat functionality. Store functionality would be optional. Preferably no build-in web browser.
> 
> The benefits of this:
> 
>     * It would perform significantly better then the Chromium-based Steam client.
> 
>     * Less likely to have significant bugs due to the much smaller code base.
> 
>     * It being open source would allow the community to build functionality upon it. For example imagine Box86 integration allowing 32 bit Steam games to be easily played on ARM-based devices running Linux.
> 
>     * Valve are the good guys and the hero of the Linux community.
> 
> 
> The downsides:
> 
>     * It will be a lot of work to build and maintain an app written in a low-level programming language like C, Rust or Vala.

There is a project, though, that remakes Steam's UI in GTK4 - called AdwSteamGTK, available on GitHub. Here's a photo from my own Steam:

![Image](https://github.com/user-attachments/assets/dd8b8d18-712f-47f3-bb13-16d57249c968)

The link to project:

https://flathub.org/apps/io.github.Foldex.AdwSteamGtk

**Comment by yioannides** [comment]
> There is a project, though, that remakes Steam's UI in GTK4 - called AdwSteamGTK

It's a good project, but it takes a heavy toll on hardware performance and hardly modifies any of the backend web page elements, like the Store home page (I have a really beefy all-AMD system and I get a noticeable 0.5'' delay on every mouse hover). It's a minimal cosmetic modification with no immediate gain, so a fully-fledged GTK client would have been far more optimal on Linux systems.

**Comment by gbschenkel** [comment]
> I would prefer a Qt/C++ or QtForPython approach instead. Using it could even help to allow the client to run on WebOS TVs for using as steam link.

As I stated before, and since SteamOS is already using KDE Plasma, would be better suggestion to move to Qt instead of GTK, which is major used with C++, but work with Python and have bindings for others languages, as Rust and Java, if they are needed, also, if the frontend devs like, they could use QML instead, which will benefit the Big Picture together with QSS (CSS but made for QML), both Widgets and QML can use Qt WebEngine for render webpages. Also KDE has Kirigami, which recently the EasyEffects ditched GTK in favor of Qt/QML/Kirigami approach.

Remembering Qt is not only for Desktop/Mobile, but it work on embedded devices too, it is used in Infotainment devices in airplanes and sea boats, as well on cars. This will be perfect for Steam/SteamOS in the different hardwares.

**Comment by Shadowblitz16** [comment]
Please use qt, lutris uses gtk, gamehub uses gtk we need a something for qt and kde.

**Comment by P1XEL0711** [comment]
> Please use qt, lutris uses gtk, gamehub uses gtk we need a something for qt and kde.

The problem with Qt in my opinion is that it is harder to theme and support for it in DEs and WMs other than KDE isn't as wide as something like GTK3.
But still, having Qt instead of relying on web components is definitely an upgrade in my book!
